optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+
+        // Two solutions for every problem.
+        Map<Integer, List<TestdataSolution>> consumedSolutions = new HashMap<>(problemCount * 2);
+
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>(problemCount);
+
+        for (int id = 0; id < problemCount; id++) {
+            List<TestdataSolution> consumedBestSolutions = Collections.synchronizedList(new ArrayList<>());
+            String solutionName = Integer.toString(id);
+            if (onlyFinalBestSolutions) {
+                jobs.add(solverManager.solve(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            } else {
+                jobs.add(solverManager.solveAndListen(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            }
+            consumedSolutions.put(id, consumedBestSolutions);
+        }
+        assertInitializedJobs(jobs);
+        assertConsumedSolutions(problemCount, onlyFinalBestSolutions, consumedSolutions);
+    }
+
+    private void assertConsumedSolutions(int problemCount, boolean onlyFinalBestSolutions, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        if (onlyFinalBestSolutions) {
+            assertConsumedFinalBestSolutions(problemCount, consumedSolutions);
+        } else {
+            assertConsumedBestSolutions(problemCount, consumedSolutions);
+        }
+    }
+
+    private void assertConsumedFinalBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(1);
+            TestdataSolution solution = consumedSolutions.get(i).get(0);
+            assertConsumedFinalBestSolution(solution);
+        }
+    }
+
+    private void assertConsumedFinalBestSolution(TestdataSolution solution) {
+        TestdataEntity entity = solution.getEntityList().get(0);
+        assertThat(entity.getCode()).isEqualTo("e1");
+        assertThat(entity.getValue().getCode()).isEqualTo("v1");
+        entity = solution.getEntityList().get(1);
+        assertThat(entity.getCode()).isEqualTo("e2");
+        assertThat(entity.getValue().getCode()).isEqualTo("v2");
+    }
+
+    private void assertConsumedBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(2);
+            assertConsumedFirstBestSolution(consumedSolutions.get(i).get(0));
+            assertConsumedFinalBestSolution(consumedSolutions.get(i).get(1));
+        }
+    }
+
+    private void assertConsumedFirstBestSolution(TestdataSolution solution) {
```

Comment:
I wouldn't extract this into a seperate method, because:
- that code isn't used twice. No DRY justifying this
- this extraction doesn't make the code easier to read (assertConsumedFirstBestSolution isn't an obvious abstraction). It makes it harder to follow the flow.